src/test/kotlin/reminder/application/ReminderSearcherTest.kt

```diff
@@ -0,0 +1,60 @@
+package reminder.application
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import reminder.domain.Reminder
+import reminder.domain.ReminderIdentifierGenerator
+import reminder.domain.ReminderRepository
+import reminder.mothers.ReminderMother
+import java.util.*
+import kotlin.test.assertEquals
+
+class ReminderSearcherTest {
+    private lateinit var reminderSaver: ReminderSaver
+    private lateinit var generator: ReminderIdentifierGenerator
+    private lateinit var reminderSearcher: ReminderSearcher
+    private lateinit var repository: ReminderRepository
+
+    @BeforeEach
+    fun setUp() {
+        repository = Mockito.mock(ReminderRepository::class.java)
+        generator = Mockito.mock(ReminderIdentifierGenerator::class.java)
+        reminderSearcher = ReminderSearcher(repository)
+        reminderSaver = ReminderSaver(repository, generator)
+    }
+
+    @Test
+    fun `Nothing is returned when valid identifier doesn't exist in the repository`() {
+        val reminder = ReminderMother.getValidReminderWithoutDescription()
+        `Assert that the reminder was not found in the repository given the identifier from`(reminder)
+    }
+
+    @Test
+    fun `Reminder is returned when identifier exists in the repository`() {
+        val reminder = ReminderMother.getValidReminderWithoutDescription()
+        `Save reminder to repository given the primitives from`(reminder)
+        `Assert that the reminder was found in the repository given the identifier from`(reminder)
+    }
+
+    private fun `Assert that the reminder was not found in the repository given the identifier from`(reminder: Reminder) {
```

Comment:
from the previous comment, this function could be removed

Reply:
Instead of removing it, what about keeping it and making it recieve an identifier so it can be easily reused on future tests? Although I am not sure if there is any other scenario that has to be tested on this use case.

Reply:
I'd remove it and if in the future need to be implemented, we'll do it. No need to keep boilerplate just in case one day's needed.